lib/queuePopulator/KafkaLogConsumer/LogConsumer.js

```diff
@@ -42,70 +46,141 @@ class LogConsumer {
-        this._consumer.on('ready', () => {
+        this._consumer.once('ready', () => {
             this._consumer.subscribe([this._topic]);
-            return done();
+            done();
         });
     }
 
     /**
-     * Waits for consumer group to rebalance
-     * @param {number} wait amount of time to wait already waited for, in milliseconds
-     * @param {Function} cb callback
+     * Offset commit callback
+     * @param {Error} err
+     * @param {Object} topicPartitions
      * @returns {undefined}
      */
-    _waitForAssignment(wait, cb) {
-        setTimeout(() => {
-            // assignements contain the partitions
-            // assigned to this consumer, they are only
-            // set once consumer group is balanced
-            const assignments = this._consumer.assignments();
-            if (assignments.length === 0) {
-                if (wait > maxDelayRebalanceMs) {
-                    this._log.error('Timeout waiting for consumer to be assigned to partitions', {
-                        method: 'LogConsumer._waitForAssignment',
-                        topic: this._topic,
-                        consumerGroupId: this._consumerGroupId,
-                    });
-                    return cb();
-                }
-                return this._waitForAssignment(wait + 2000, cb);
+    _onOffsetCommit(err, topicPartitions) {
+        if (err) {
+            // NO_OFFSET is a "soft error" meaning that the same
+            // offset is already committed, which occurs because of
+            // auto-commit (e.g. if nothing was done by the producer
+            // on this partition since last commit).
+            if (err.code === kafka.CODES.ERRORS.ERR__NO_OFFSET) {
+                return undefined;
             }
-            return cb();
-        }, 2000);
+            this._log.error('Error committing offsets to kafka', {
+                method: 'LogConsumer._onOffsetCommit',
+                errorCode: err,
+                topicPartitions,
+                groupId: this._groupId
+            });
+            return undefined;
+        }
+        this._pendingCommit = false;
+        this._log.debug('Offsets committed', {
+            method: 'LogConsumer._onOffsetCommit',
+            topicPartitions,
+            groupId: this._groupId
+        });
+        return undefined;
     }
 
     /**
-     * Queries last commited partition offsets
-     * and stores them
-     * @param {Function} cb callback
-     * @returns {undefined}
+     * @param {kafka.KafkaError} err Rebalance event
+     * @param {TopicPartition[]} assignment List of (un)assigned partitions
+     * @returns {void}
      */
-    _storeCurrentOffsets(cb) {
-        this._consumer.committed(5000, (err, topicPartitions) => {
-            if (err) {
-                this._log.error('Error while getting offsets', {
-                    method: 'LogConsumer._storeCurrentOffsets',
+    _onRebalance(err, assignment) {
```

Comment:
should this update metrics: `KafkaBacklogMetrics.onRebalance(this._topic, this._groupId, status)` ?